src/main/java/stelnet/storage/data/SharedData.java

```diff
@@ -11,14 +11,16 @@
 import stelnet.helper.LogHelper;
 import stelnet.storage.ButtonManager;
 import stelnet.storage.FilterManager;
-import stelnet.ui.GridData;
+import stelnet.ui.Group;
 import stelnet.ui.Heading;
 import stelnet.ui.Renderable;
-import stelnet.ui.ScrollableStack;
 import stelnet.ui.Size;
 import stelnet.ui.Stack;
 
-public abstract class SharedData implements GridData {
+public abstract class SharedData {
```

Comment:
You could take this class as a good exercise of getting rid of the inheritance and using composition instead. Try looking at some of the design patterns here. Possibly decorator pattern.

https://refactoring.guru/design-patterns/catalog

Also consider renaming this class and all other classes that inherit it. "SharedData" would imply repository or entity/dto. However actually it is a table that has columns and a header. Perhaps DataTableDialog or DataTable would be better.

Reply:
Made #26